router/usecase/pools/pool_factory.go

```diff
@@ -96,20 +96,47 @@ func newRoutableCosmWasmPool(pool sqsdomain.PoolI, cosmWasmConfig domain.CosmWas
 	cosmwasmPool, ok := chainPool.(*cwpoolmodel.CosmWasmPool)
 	if !ok {
 		return nil, domain.FailedToCastPoolModelError{
-			ExpectedModel: poolmanagertypes.PoolType_name[int32(poolmanagertypes.Balancer)],
+			ExpectedModel: poolmanagertypes.PoolType_name[int32(poolmanagertypes.CosmWasm)],
 			ActualModel:   poolmanagertypes.PoolType_name[int32(poolType)],
 		}
 	}
 
+	// Check if the pool is a transmuter pool with alloyed assets
+	model := pool.GetSQSPoolModel().CosmWasmPoolModel
+	if model != nil {
+		// since v2, we introduce concept of alloyed assets but not yet actively used
+		// since v3, we introduce concept of normalization factor
+		// `routableAlloyTransmuterPoolImpl` is v3 compatible
+		if model.IsAlloyTransmuter() {
+			spreadFactor := pool.GetSQSPoolModel().SpreadFactor
+			balances := pool.GetSQSPoolModel().Balances
+
+			if model.Data.AlloyTransmuter == nil {
+				return nil, domain.AlloyTransmuterDataMissingError{PoolId: pool.GetId()}
+			}
+
+			return &routableAlloyTransmuterPoolImpl{
+				ChainPool:           cosmwasmPool,
+				AlloyTransmuterData: model.Data.AlloyTransmuter,
+				Balances:            balances,
+				TokenOutDenom:       tokenOutDenom,
+				TakerFee:            takerFee,
+				SpreadFactor:        spreadFactor,
+			}, nil
+		}
+	}
```

Comment:
I think we should still have the ability to control whether we support an alloyed transmitter by having a config for them somewhere [here](https://github.com/osmosis-labs/sqs/blob/7f9fdda008ac7dd6ab599caec64622d375543d44/config.json#L20).

In previous cw pool releases, we had several unexpected issues arise so controlling what is supported by our router proved to be useful.